openapi/tests/openapi_integration/test_geo_filter.py

```diff
@@ -0,0 +1,289 @@
+                                        {"lon": -180.0, "lat": -90.0},
+                                        {"lon": 180.0, "lat": -90.0},
+                                        {"lon": 180.0, "lat": 90.0},
+                                        {"lon": -180.0, "lat": 90.0},
+                                        {"lon": -180.0, "lat": -90.0},
+                                    ]
+                            },
+                            "interiors": [],
+                        },
+                    }
+                ]
+            },
+        },
+    )
+    assert response.ok
+
+    json = response.json()
+    assert len(json["result"]) == 3
+
+    ids = [x["id"] for x in json["result"]]
+    assert 1 in ids
+    assert 3 in ids
+    assert 4 in ids
+
+    # a polygon with interior
+    response = request_with_validation(
+        api="/collections/{collection_name}/points/search",
+        method="POST",
+        path_params={"collection_name": collection_name},
+        body={
+            "vector": [0.2, 0.1, 0.9, 0.7],
+            "limit": 3,
+            "filter": {
+                "must": [
+                    {
+                        "key": "location",
+                        "geo_polygon": {
+                            "exterior":{
+                                    "points": [
+                                        {"lon": -70.0, "lat": -70.0},
+                                        {"lon": 60.0, "lat": -70.0},
+                                        {"lon": 60.0, "lat": 60.0},
+                                        {"lon": -70.0, "lat": 60.0},
+                                        {"lon": -70.0, "lat": -70.0},
+                                    ]
+                            },
+                            "interiors": [
+                                {
+                                    "points": [
+                                        {"lon": -65.0, "lat": -65.0},
+                                        {"lon": 0.0, "lat": -65.0},
+                                        {"lon": 0.0, "lat": 0.0},
+                                        {"lon": -65.0, "lat": 0.0},
+                                        {"lon": -65.0, "lat": -65.0},
+                                    ]
+                                }
+                            ]
+                        },
+                    }
+                ]
+            },
+        },
+    )
+    assert response.ok
+
+    json = response.json()
+    assert len(json["result"]) == 1
+
+    ids = [x["id"] for x in json["result"]]
+    assert 1 in ids
+
+    # multiple polygons
+    response = request_with_validation(
+        api="/collections/{collection_name}/points/search",
+        method="POST",
+        path_params={"collection_name": collection_name},
+        body={
+            "vector": [0.2, 0.1, 0.9, 0.7],
+            "limit": 3,
+            "filter": {
+                "should": [
```

Comment:
maybe you could copy paste this query with `must` to show it returns an empty set.

Reply:
added.